spec/extensions/git/base_spec.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Git::Base do
+  let(:repo) { Git.open('.') }
+
+  describe '#merge_base' do
+    subject { repo.merge_base('HEAD^', 'HEAD') }
+
+    let(:prehead) { repo.gcommit('HEAD^').sha }
+
+    it { is_expected.to eq prehead }
+  end
+end
```

Comment:
I've modified this spec a bit however you should modify it more when addressing my comment above

Reply:
adding `.sha` to subject is not that much :)

Reply:
why do you need `sha`? Comparing by objects doesn't work?